### DISCUSSION TITLE
PImcore REST API

### DISCUSSION THREAD

**Discussion by bhagya7977** [pimcore/pimcore#16556] [Q&A / Support]
Hello guys,
how to implement the REST API in Pimcore (11.0.2) and how to get the recently updated products in the API response ?

**Comment by kingjia90** [comment]
Please have a look to https://github.com/pimcore/data-hub and https://pimcore.com/docs/platform/Datahub_Simple_Rest/Filtering_and_Paging (enteprise), should be what you are looking for

**Reply by bhagya7977** [reply]
In my DataObject class, there are no modification date  & CreatedDate fields, how do i filter the listing ?

**Reply by kingjia90** [reply]
Those are system fields, you don't have to specify/create them in the UI, but it is available in the APIs

**Reply by bhagya7977** [reply]
In my Pimcore instance, I have created a Data object class with a name, SKU,  price, and Qty fields under Product Class.    
I have created 5 products using the Product class .  
I have created an new Controller method  getProductList()    and  declared the route under    config/routes.yaml file 


`admin_pimcoredatahub_config:
    
path: /pimcore-datahub-webservics/getProductList

   defaults: { _controller: App\Controller\ProductRestController::getProductList }`


The method **getProductList()::**

`$products = new \Pimcore\Model\DataObject\DemoProduct\Listing();      foreach ($products as $key => $blog) {
            $data[] = array(
                "sku" => $blog->getSku(),
                "name" => $blog->getName(),
                "price" => $blog->getPrice(),
                "status" => $blog->getStatus()},`

so when i am hitting the    API   <BASEURL>/pimcore-datahub-webservics/getProductList 

getting the output format 

`[{"sku":"24-MB01","name":"Joust Duffle Bag-test","price":34.09,"status":1,"qty":135},{"sku":"24-MB04-new","name":"Strive Shoulder Pack","price":32,"status":1,"qty":138},{"sku":"24-MB03-45","name":"Crown Summit Backpack","price":38,"status":1,"qty":190},{"sku":"24-MB057","name":"Wayfarer Messenger Bag new product","price":50,"status":1,"qty":300},{"sku":"24-MB06","name":"Rival Field Messenger","price":45,"status":1,"qty":150}]`

so what customization i need to update to get recently updated product list in the getProductList() method?

**Reply by kingjia90** [reply]
From there should be easy, setOrderKey and limit etc..

See https://pimcore.com/docs/platform/Pimcore/Objects/Working_with_PHP_API/#object-listings

**Reply by bhagya7977** [reply]
sorry for late reply , As its working , i have utilized the parameter modificationDate  with method setOrderKey .

**Comment by bhagya7977** [comment]
issue resolved
